### ISSUE TITLE
[BUG] Authorization error when publishing midterm_project to github

### ISSUE THREAD

**Issue by just-meng** [datalad-handbook/book#850]
### Current behavior

Hi all,

I am following this amazing handbook to learn how to use DataLad for my project. Here is a bug I encounter when trying to create a git repo for the "midterm_project" in section 6.3.

> jiame@RAZOR MINGW64 ~/Projects/DataLad-101/midterm_project (adjusted/master(unlocked))
> $ datalad create-sibling-github -d . --github-login ghp_Twu2B2oDAWOFWEKNSAQQD8L2fpP2SDcFt midtermproject
> [WARNING] Cannot determine authorization token for githubloginarg
> [ERROR] Authorization required for GitHub, cannot find token for a credential githubloginarg. (ValueError)

Thanks for your help,
Meng

### Operating system

- Windows 11
- git-bash

### DataLad information

datalad-0.17.1

### Browser

- Chrome

**Comment by adswa** [comment]
Thanks a lot for opening this issue! The handbook is somewhat wrong, the ``--github-login`` parameter was deprecated in the last release. I'll create an issue to remove this from the handbook. But theoretically it should continue to work for a while still, I'd be interested in finding out what went wrong.

Let's get you able to publish to GitHub. In theory, with your version of DataLad, you should get an interactive prompt saying something like "An access token is required for github.com, visit ... to generate a token" when you run this command the first time - did you get that? And if so, what did you do?

**Comment by just-meng** [comment]
I never got any interactive promt running either the above command or the one without --github-login:

`$ datalad create-sibling-github -d .  midtermproject`

Always got the same warning and error message.

**Comment by adswa** [comment]
Thanks for clarifying! I just found a bug that might explain this. Could you maybe paste the output of a command that lists your installed python packages? The exact command is dependent on how you install packages - if you install Python packages via conda, you could run ``conda list``, if you use pip, you could run ``pip freeze``

**Comment by just-meng** [comment]
$ pip freeze
annexremote==1.6.0
boto==2.49.0
certifi==2022.6.15
chardet==4.0.0
charset-normalizer==2.1.0
colorama==0.4.5
cycler==0.11.0
datalad==0.17.1
datalad-installer==0.9.0
distro==1.7.0
fasteners==0.17.3
fonttools==4.34.4
humanize==4.2.3
idna==3.3
iso8601==1.0.2
joblib==1.1.0
keyring==23.6.0
keyrings.alt==4.1.0
kiwisolver==1.4.3
matplotlib==3.5.2
msgpack==1.0.4
numpy==1.23.1
packaging==21.3
pandas==1.4.3
patool==1.12
Pillow==9.2.0
platformdirs==2.5.2
pyparsing==3.0.9
python-dateutil==2.8.2
python-gitlab==3.6.0
pytz==2022.1
pywin32-ctypes==0.2.0
requests==2.28.1
requests-toolbelt==0.9.1
scikit-learn==1.1.1
scipy==1.8.1
seaborn==0.11.2
simplejson==3.17.6
six==1.16.0
sklearn==0.0
threadpoolctl==3.1.0
tqdm==4.64.0
urllib3==1.26.10
Whoosh==2.7.4
****

**Comment by adswa** [comment]
thx! Could you please try the following:
- install ``datalad-next`` - you can get it via ``pip install datalad-next``, it should be the just released version 0.4.1
- run ``datalad create-sibling-github midtermproject`` (i.e., no token at all specified). You should see a prompt similar to the one below:
![Screenshot from 2022-07-14 14-58-35](https://user-images.githubusercontent.com/29738718/178988283-5a301a78-7c52-40fa-8147-c58da0907a5d.png)

**Comment by just-meng** [comment]
I retried with windows cmd and got the promt. Went to github and generated a new token, pasted into the cmd and got another error:

![image](https://user-images.githubusercontent.com/108482432/178995514-acdb5bc9-2028-4f44-be90-6446059d406c.png)

Also the prompt was only show for the first time I ran the command.

**Comment by adswa** [comment]
Well, that's at least progress. Sorry this is dragging along so slowly. If you are available in 15 minutes we can video call in-person in the [DataLad Office hour]( https://us02web.zoom.us/j/85705522690?pwd=VnVPRUJlVlNSYVFOYkF5OTFzSkJ4dz09), it might be faster like that. 
(answering in a later comment, but wanted to get the zoom link out)

**Comment by adswa** [comment]
Wohoo! Congrats!
Sorry it took so long, and well done for solving this. Now that the right credential is in the credential manager, future calls should work until the token expires. 
I suspect that the "400" Status code in CMD comes from a malformatted token - that's not your fault or a wrong token, just terminal + CTRL V. I saw something similar in my test environment but had hoped this would be because I need to connect remotely to a windows test environment.

You can set tokens non-interactively from the command line with ``datalad credentials set [name] secret=<your-secret-here>``. Afterwards, you can run commands like ``create-sibling-github`` with the ``--credential [name]`` specification to point them to the right credential. I created an issue about the non-interactive prompt in DataLad core: https://github.com/datalad/datalad/issues/6838

**Comment by adswa** [comment]
This is great! Happy to help. I will close this issue, and opened #851 on removing the deprecated parameter.

**Comment by just-meng** [comment]
Hi Adina,

I'm running into a very similar problem again and none of our previous solutions works. 

So I have a datalad dataset hosted on gin, and want to clone it onto a new PC. All installations are fresh and followed the datalad handbook. After adding the SSH key to the repo, I created a personal-access-token, and tried running the following in gitbash/windows cmd:

`datalad credentials set [name] token=<personal-access-token>
`

Got the following complaint:
`datalad: Unknown command 'credentials'.  See 'datalad --help'.`

My system:

- datalad 0.17.2
- git-annex version: 8.20211117-gc3af94eff
- git version 2.37.2.windows.2
- Windows 10

**Comment by adswa** [comment]
Hi @just-meng! Can you insstall ``datalad next`` on your new PC as well? This package currently provides the ``credentials`` command.

**Comment by just-meng** [comment]
I did pip install datalad next and open a new gitbash, still shows the same problem.

**Comment by just-meng** [comment]
My bad! It's `pip install datalad-next`. Now it works! :D Thank you!

**Comment by adswa** [comment]
ah and sorry for the typo from my side :D
